src/createServer.js

```diff
@@ -1,11 +1,186 @@
 'use strict';
 
-// const express = require('express');
+const express = require('express');
 
 function createServer() {
   // Use express to create a server
   // Add a routes to the server
   // Return the server (express app)
+  const app = express();
+
+  let usersData = [];
+  let exprensesData = [];
+
+  app.delete('/users/:userId', (req, res) => {
+    const { userId } = req.params;
+    const filteredData = usersData.filter(user => user.id !== +userId);
+
+    if (filteredData.length === usersData.length) {
+      res.sendStatus(404);
+    } else {
+      usersData = filteredData;
+      res.status(204);
+      res.send(usersData);
+    }
+  });
+
+  app.patch('/users/:userId', express.json(), (req, res) => {
+    const { userId } = req.params;
+    const foundUserId = usersData.find(user => user.id === +userId);
+
+    if (!foundUserId) {
+      res.sendStatus(404);
+    } else {
+      const { name } = req.body;
+
+      Object.assign(foundUserId, { name });
+
+      res.send(foundUserId);
+    }
+  });
+
+  app.get('/users', (req, res) => {
+    if (!usersData) {
+      res.send([]);
+    } else {
+      res.setHeader('Content-Type', 'application/json');
+      res.status(200);
+      res.send(usersData);
+    }
+  });
+
+  app.get('/users/:userId', (req, res) => {
+    const { userId } = req.params;
+    const foundUserId = usersData.find(user => user.id === +userId);
+
+    if (!foundUserId) {
+      res.status(404);
+    }
+
+    res.send(foundUserId);
+  });
```

Comment:
Also u should check is your request correct

```suggestion
  app.get('/users/:userId', (req, res) => {
    const { userId } = req.params;
    
    if (isNaN(+userId)) {
      res.sendStatus(400);

      return;
    }
    
    const foundUserId = usersData.find(user => user.id === +userId);

    if (!foundUserId) {
      res.status(404);
    }

    res.send(foundUserId);
  });
```

![image](https://user-images.githubusercontent.com/107215656/197489027-3fb6455a-b1a3-4d8a-9b79-074af0c3a3e8.png)